### PR TITLE
Support Kafka as a source and as a sink

### PR DESCRIPTION
**Pull request by hayssams**
## Summary
This PR support Kafka as a view, source or sink.
We do not rely on Spark Structured Streaming since we need to use Kafka topics as a source/view in Spark Batch jobs too.
Only JSON messages are supported for now.

Messages are read and last read offsets are stored in a specific topic. This will allow the next batch to start at the correct offset.

An improvement would be to read a maximum number of messages / batch to avoid too large dataframes.

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        .action((x, c) => c.copy(topic = x))
+        .text("Read/Write format eq : parquet, json, csv ... Default to parquet.")
+        .optional(),
+      opt[String]("path")
+        .action((x, c) => c.copy(path = Some(x)))
+        .text("Source file for load and target file for store")
+        .required(),
+      opt[String]("mode")
+        .action((x, c) => c.copy(mode = SaveMode.valueOf(x)))
+        .text("Source file for load and target file for store")
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala:48]: update text

File: src/main/scala/com/ebiznext/comet/job/ingest/KafkaIngestionJob.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+)(implicit settings: Settings)
+    extends JsonIngestionJob(domain, schema, types, path, storageHandler, schemaHandler, options) {
+
+  var offsets: List[(Int, Long)] = null
+
+  val kafkaTopicUtils = new KafkaTopicUtils(settings.comet.kafka)
+
+  private val topicConfig: Settings.KafkaTopicOptions = settings.comet.kafka.topics(schema.name)
+
+  /** Load dataset using spark csv reader and all metadata. Does not infer schema.
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/job/ingest/KafkaIngestionJob.scala:58]: update scaladoc

**Review by mhdkassir — APPROVED** [review]
LGTM as a first version of Kafka integration, except for some minor changes (documentation, unused code). I think the next step should be exception handling in `KafkaTopicUtils`.

File: src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package com.ebiznext.comet.job.index.kafkaload
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala:1]: code format need to be applied on this filel, https://github.com/ebiznext/comet-data-pipeline/pull/434/checks?check_run_id=1792588002

File: src/main/scala/com/ebiznext/comet/job/ingest/KafkaIngestionJob.scala (hunk 2 above)
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/ingest/KafkaIngestionJob.scala:58]: Done

File: src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala (hunk 3 above)
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/index/kafkaload/KafkaJobConfig.scala:1]: Done

File: src/main/scala/com/ebiznext/comet/utils/kafka/KafkaTopicUtils.scala [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  createTopicIfNotPresent(
+    new NewTopic(
+      "comet_offsets",
+      cometOffsetsConfig.partitions,
+      cometOffsetsConfig.replicationFactor
+    ),
+    Map("cleanup.policy" -> "compact")
+  )
+
+  def teardown(): Unit = client.close()
```
[the remark this answers is not in the document]
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/utils/kafka/KafkaTopicUtils.scala]: Renamed it to close and converted the class to an auto closable resource.

**Pull request merged**